targets/unity-v2/source/ExampleTestProject/Assets/Testing/Tests/Client/ClientApiTests.cs

```diff
@@ -499,6 +498,95 @@ private void WriteEventCallback(WriteEventResponse result)
+        /// </summary>
+        [UUnitTest]
+        private void UnityWebRequestTimeOutTest(UUnitTestContext testContext)
+        {
+            originalTimeout = PlayFabSettings.RequestTimeout;
+            originalWebRequestType = PlayFabSettings.RequestType;
+
+            PlayFabSettings.RequestTimeout = 0; // force this timeout
+            PlayFabSettings.RequestType = WebRequestType.HttpWebRequest;
+
+            var loginRequest = new LoginWithCustomIDRequest
+            {
+               CustomId = PlayFabSettings.BuildIdentifier,
+               CreateAccount = true,
+            };
+
+            clientInstance.LoginWithCustomID(loginRequest, LoginTimeout, LoginTimeoutError);
+        }
+
+        private void LoginTimeout(LoginResult result)
+        {
+           RestoreTimeoutSettings();
+           ((UUnitTestContext)result.CustomData).Fail("This test was expected to time out");
+        }
+
+        private void LoginTimeoutError(PlayFabError result)
+        {
+           RestoreTimeoutSettings();
+           if(result.ErrorMessage.Contains("Timeout"))
+           {
+               ((UUnitTestContext)result.CustomData).EndTest(UUnitFinishState.PASSED, "This test was expected to time out");
+           }
+           else
+           {
+               RestoreTimeoutSettings();
+               ((UUnitTestContext)result.CustomData).Fail("This test was expected to time out");
+           }
+        }
+
+        [UUnitTest]
+        private void UnityWebRequestTimeOutIgnoredTest(UUnitTestContext testContext)
+        {
+            originalTimeout = PlayFabSettings.RequestTimeout;
+            originalWebRequestType = PlayFabSettings.RequestType;
+
+            PlayFabSettings.RequestTimeout = -1;
+            PlayFabSettings.RequestType = WebRequestType.UnityWebRequest;
+
+            var loginRequest = new LoginWithCustomIDRequest
+            {
+               CustomId = PlayFabSettings.BuildIdentifier,
+               CreateAccount = true,
+            };
+
+            clientInstance.LoginWithCustomID(loginRequest, LoginTimeoutIgnored, LoginTimeoutIgnoredError, testContext);
+        }
+
+        private void LoginTimeoutIgnored(LoginResult result)
+        {
+            RestoreTimeoutSettings();
+            ((UUnitTestContext)result.CustomData).EndTest(UUnitFinishState.PASSED, "This test was expected to not time out");
+        }
+
+        private void LoginTimeoutIgnoredError(PlayFabError result)
+        {
+           RestoreTimeoutSettings();
+           if(result.ErrorMessage.Contains("Timeout"))
+           {
+               ((UUnitTestContext)result.CustomData).EndTest(UUnitFinishState.TIMEDOUT, "This test was not expected to time out");
+           }
+           else
+           {
+               RestoreTimeoutSettings();
+               ((UUnitTestContext)result.CustomData).Fail("This test was expected to see this error " + result.ErrorMessage);
+           }
+        }
+
+        void RestoreTimeoutSettings()
```

Comment:
I think RestoreTimeoutSettings should be in the class "TearDown()" method, not a mid-test method.
There's lots of reasons why the test could fail, or except, or crash, or whatever, and TearDown has a very high chance of still being called. But if it doesn't get called in the test for some reason, it just won't happen.

Likewise, I think class StartUp() is the time to save the timeout.